doc: release process chainparams updates

jonatack opened the pull request:
Release process updates, fixes and clarifications regarding updating the chainparams:
 
- add missing references to signet
- specify specify blockchain/chainstate units, reduce repetition
- exclude huge files for m_assumed_blockchain_size on mainnet
- rewrite overhead note to be more actionable
- reorganize the chainparams section to reduce repetition
- specify which chains need to be updated
- use 4096 blocks and getbestblockhash for getchaintxstats
- clarify how to update defaultAssumeValid and nMinimumChainWork

jonatack committed:
Add missing references to signet in the release process

jonatack committed:
Release process: specify blockchain/chain_state units, reduce repetition

laanwj commented on doc/release-process.md:
```diff
@@ -300,15 +300,15 @@ cat "$VERSION"/*/all.SHA256SUMS.asc > SHA256SUMS.asc
 Notes:
 - When taking the size for `m_assumed_blockchain_size`, there's no need to exclude the `/chainstate` directory since it's a guideline value and an overhead will be added anyway.
-- The expected overhead for growth may change over time, so it may not be the same value as last release; pay attention to that when changing the variables.
+- The expected overhead for growth may change over time, so it may not be the same value as the previous release; pay attention to that when updating the variables.
```
BTW. this sentence (not introduced here) is completely un-actionable for someone executing the steps. Pay attention to what? It needs to be more specific, or maybe just left out.

jonatack replied:
Reworked to be more specific and actionable.

jonatack replied:
(Good point.)

laanwj commented on doc/release-process.md:
```diff
@@ -28,8 +28,8 @@ Release Process
 * Update hardcoded [seeds](/contrib/seeds/README.md), see [this pull request](https://github.com/bitcoin/bitcoin/pull/7415) for an example.
-* Update [`src/chainparams.cpp`](/src/chainparams.cpp) m_assumed_blockchain_size and m_assumed_chain_state_size with the current size plus some overhead (see [this](#how-to-calculate-assumed-blockchain-and-chain-state-size) for information on how to calculate them).
-* Update [`src/chainparams.cpp`](/src/chainparams.cpp) chainTxData with statistics about the transaction count and rate. Use the output of the `getchaintxstats` RPC, see
+* Update [`src/chainparams.cpp`](/src/chainparams.cpp) m_assumed_blockchain_size and m_assumed_chain_state_size for mainnet, testnet, and signet with the current size plus some overhead (see [this](#how-to-calculate-assumed-blockchain-and-chain-state-size) for information on how to calculate them).
```
I've also thought about nesting the three concerning chainparam under one item somehow. There's a lot of repetition.

jonatack replied:
Good point, reorganized.

jonatack committed:
Reorganize release process chainparams section to reduce repetition

jonatack committed:
Clarify release process overhead note to be more actionable

jonatack commented:
Updated per @laanwj feedback (good ideas, thanks).

prayank23 reviewed (approved):
ACK https://github.com/bitcoin/bitcoin/pull/24424/commits/bba0655c6fd6908edc1a43146ec52fbf95124a68

Updates improve documentation and I could not find any issues.

laanwj commented:
Other things to add maybe:
- Use 4096 blocks for `getchaintxstats` https://github.com/bitcoin/bitcoin/pull/24418#discussion_r812988911
- Try to use the same block for `getchaintxstats` and `getblockheader` (e.g. for the chainTxData and nMinimumChainWork/defaultAssumeValid steps)

Sjors commented:
It might also be worth writing a script to fetch these values based on a given height, both for reviewers and the maintainer.

jonatack committed:
Release process: use 4096 blocks and getbestblockhash for getchaintxstats

fanquake commented:
Some of the commits here can be dropped. i.e 4b5b011c1d0eba8a432e067c28278f22c18689ff adds a number of `for mainnet, testnet, and signet`, but then they are just deleted in 5099f23b7b29e869af463fa53a3442be011bf039.

jonatack commented:
> Some of the commits here can be dropped. i.e 4b5b011c1d0eba8a432e067c28278f22c18689ff adds a number of `for mainnet, testnet, and signet`, but then they are just deleted in 5099f23b7b29e869af463fa53a3442be011bf039.

The commit specifying the chains can be moved after the reorganization. I'm reworking to incorporate the latest feedback here and the other items I've noticed.

laanwj commented:
> It might also be worth writing a script to fetch these values based on a given height, both for reviewers and the maintainer.

Yes, automation would be even better. There's quite some scope for reducing human error-proneness and pitfalls here. That said, documenting what we really want to be done is the first step.

laanwj commented on doc/release-process.md:
```diff
@@ -300,15 +303,15 @@ cat "$VERSION"/*/all.SHA256SUMS.asc > SHA256SUMS.asc
-- For `testnet` -> Take the size of the `/testnet3` directory.
+To calculate `m_assumed_blockchain_size`, take the size in GiB of these directories:
+- For `mainnet` -> the data directory, excluding the `/regtest`, `/signet`, and `/testnet3` directories
+- For `testnet` -> `/testnet3`
```
We also want to exclude overly large files in the root directory of the network, e.g. a huge debug.log as tripped up in #24418 :smile: I'm not sure what is a good calculation, but maybe a simple reminder would do.

jonatack replied:
Added in reminder in 39f85b4

jonatack committed:
Clarify in release process how to update defaultAssumeValid/nMinimumChainWork

jonatack committed:
Specify in release process which chains need to be updated

laanwj committed:
Release process: exclude huge files for mainnet m_assumed_blockchain_size

prayank23 reviewed (approved):
reACK https://github.com/bitcoin/bitcoin/pull/24424/commits/9560757adc90938b54148e3921aacfaa4aa51be6

laanwj commented on doc/release-process.md:
```diff
@@ -300,15 +306,15 @@ cat "$VERSION"/*/all.SHA256SUMS.asc > SHA256SUMS.asc
-- For `testnet` -> Take the size of the `/testnet3` directory.
+To calculate `m_assumed_blockchain_size`, take the size in GiB of these directories:
+- For `mainnet` -> the data directory, excluding the `/regtest`, `/signet`, and `/testnet3` directories and any overly large files, e.g. a huge `debug.log`
+- For `testnet` -> `/testnet3`
```
No signet for `m_assumed_blockchain_size`? I mean, I guess it's way too small to be relevant at this point
```
m_assumed_blockchain_size = 1;
m_assumed_chain_state_size = 0;
```
But you do add it for `m_assumed_chain_state_size` below.

jonatack replied:
Good catch! Fixed.

laanwj commented:
ACK 9560757adc90938b54148e3921aacfaa4aa51be6
modulo signet question

brunoerg commented on doc/release-process.md:
```diff
@@ -300,15 +306,15 @@ cat "$VERSION"/*/all.SHA256SUMS.asc > SHA256SUMS.asc
-- For `mainnet` -> Take the size of the data directory, excluding `/regtest` and `/testnet3` directories.
-- For `testnet` -> Take the size of the `/testnet3` directory.
+To calculate `m_assumed_blockchain_size`, take the size in GiB of these directories:
+- For `mainnet` -> the data directory, excluding the `/regtest`, `/signet`, and `/testnet3` directories and any overly large files, e.g. a huge `debug.log`
```
What size can be considered a huge `debug.log`?

jonatack replied:
See https://github.com/bitcoin/bitcoin/pull/24424#discussion_r815820014.

jonatack replied:
(Happy to update if you have a suggestion.)

brunoerg replied:
Np, sounds good.

brunoerg reviewed (approved):
ACK 9560757adc90938b54148e3921aacfaa4aa51be6

jonatack commented:
Thanks for reviewing! Addressed @laanwj's feedback, fixed an indentation, and reordered the directories to exclude in the `m_assumed_blockchain_size` mainnet calculation from large to small instead of small to large.

<details><summary><code>git diff 9560757 74743ad9</code></summary><p>

```diff
--- a/doc/release-process.md
+++ b/doc/release-process.md
@@ -31,8 +31,8 @@ Release Process
 * Update the following variables in [`src/chainparams.cpp`](/src/chainparams.cpp) for mainnet, testnet, and signet:
   - `m_assumed_blockchain_size` and `m_assumed_chain_state_size` with the current size plus some overhead (see
     [this](#how-to-calculate-assumed-blockchain-and-chain-state-size) for information on how to calculate them).
-    - The following updates should be reviewed with `reindex-chainstate` and `assumevalid=0` to catch any defect
-      that causes rejection of blocks in the past history.
+  - The following updates should be reviewed with `reindex-chainstate` and `assumevalid=0` to catch any defect
+    that causes rejection of blocks in the past history.
   - `chainTxData` with statistics about the transaction count and rate. Use the output of the `getchaintxstats` RPC with an
     `nBlocks` of 4096 (28 days) and a `bestblockhash` of RPC `getbestblockhash`; see
     [this pull request](https://github.com/bitcoin/bitcoin/pull/20263) for an example. Reviewers can verify the results by running
@@ -307,8 +307,9 @@ Both variables are used as a guideline for how much space the user needs on thei
 Note that all values should be taken from a **fully synced** node and have an overhead of 5-10% added on top of its base value.
 
 To calculate `m_assumed_blockchain_size`, take the size in GiB of these directories:
-- For `mainnet` -> the data directory, excluding the `/regtest`, `/signet`, and `/testnet3` directories and any overly large files, e.g. a huge `debug.log`
+- For `mainnet` -> the data directory, excluding the `/testnet3`, `/signet`, and `/regtest` directories and any overly large files, e.g. a huge `debug.log`
 - For `testnet` -> `/testnet3`
+- For `signet` -> `/signet`
 
 To calculate `m_assumed_chain_state_size`, take the size in GiB of these directories:
 - For `mainnet` -> `/chainstate`
```
</p></details>

brunoerg reviewed (approved):
re-ACK 74743ad90590708b46a8bc8cb9cefedb66471306

theStack commented:
Concept ACK

laanwj commented:
ACK 74743ad90590708b46a8bc8cb9cefedb66471306
